index.php

```diff
@@ -77,17 +82,82 @@ function dataDisplay($dataInput)
+        }
+        elseif ($_GET['sort'] === 'publication_date')
+        {
+            $query = $db->prepare('SELECT `image`, `title`, `forename`, `surname`, `publication_date`, `rating`, `genre_1` FROM `books` INNER JOIN `authors` ON `authors`.`id` = `author_id` ORDER BY `publication_date` DESC ');
+            $query->execute();
+            $sortBooks = $query->fetchAll();
+        }
+        elseif ($_GET['sort'] === 'genre')
+        {
+            $query = $db->prepare('SELECT `image`, `title`, `forename`, `surname`, `publication_date`, `rating`, `genre_1` FROM `books` INNER JOIN `authors` ON `authors`.`id` = `author_id` ORDER BY `genre_1` DESC ');
+            $query->execute();
+            $sortBooks = $query->fetchAll();
+        }
+        elseif ($_GET['sort'] === 'rating')
+        {
+            $query = $db->prepare('SELECT `image`, `title`, `forename`, `surname`, `publication_date`, `rating`, `genre_1` FROM `books` INNER JOIN `authors` ON `authors`.`id` = `author_id` ORDER BY `rating` DESC ');
+            $query->execute();
+            $sortBooks = $query->fetchAll();
+        }
+    }
+    elseif ($_GET['sortBy'] === 'asc')
+    {
+        if ($_GET['sort'] === 'title')
+        {
+            $query = $db->prepare('SELECT `image`, `title`, `forename`, `surname`, `publication_date`, `rating`, `genre_1` FROM `books` INNER JOIN `authors` ON `authors`.`id` = `author_id` ORDER BY `title` ASC ');
+            $query->execute();
+            $sortBooks = $query->fetchAll();
+        }
+        elseif ($_GET['sort'] === 'author')
+        {
+            $query = $db->prepare('SELECT `image`, `title`, `forename`, `surname`, `publication_date`, `rating`, `genre_1` FROM `books` INNER JOIN `authors` ON `authors`.`id` = `author_id` ORDER BY `surname` DESC ');
+            $query->execute();
+            $sortBooks = $query->fetchAll();
+        }
+        elseif ($_GET['sort'] === 'publication_date')
+        {
+            $query = $db->prepare('SELECT `image`, `title`, `forename`, `surname`, `publication_date`, `rating`, `genre_1` FROM `books` INNER JOIN `authors` ON `authors`.`id` = `author_id` ORDER BY `publication_date` ASC ');
+            $query->execute();
+            $sortBooks = $query->fetchAll();
+        }
+        elseif ($_GET['sort'] === 'genre')
+        {
+            $query = $db->prepare('SELECT `image`, `title`, `forename`, `surname`, `publication_date`, `rating`, `genre_1` FROM `books` INNER JOIN `authors` ON `authors`.`id` = `author_id` ORDER BY `genre_1` ASC ');
+            $query->execute();
+            $sortBooks = $query->fetchAll();
+        }
+        elseif ($_GET['sort'] === 'rating') {
+            $query = $db->prepare('SELECT `image`, `title`, `forename`, `surname`, `publication_date`, `rating`, `genre_1` FROM `books` INNER JOIN `authors` ON `authors`.`id` = `author_id` ORDER BY `rating` ASC ');
+            $query->execute();
+            $sortBooks = $query->fetchAll();
+        }
+    }
+
```

Comment:
feels like you have repeated code a lot here but I am unsure if that is avoidable/ could not tell you how to do it differently.